### ISSUE TITLE
Investigate the effiency of the chain download pipeline

### ISSUE THREAD

**Issue by jframe** [hyperledger/besu#7696]
Investigate the efficiency of the chain download pipeline. There could be some pauses in the pipeline due to waiting on stages to complete.

**Comment by ahamlat** [comment]
Yes, agreed. The pipeline is a set of steps that are executing in parallel and in a certain order. Each stage is dependant from the stage before and the stage after as we can see in the diagram below. We need to review the metrics to be able to spot wait time on each step on either when putting data to the next step or when reading data from the step before.
In a perfect configuration, each step should be doing work all the time, not blocking on either the step before or the step after.

![image](https://github.com/user-attachments/assets/8ba4f1e2-cb5e-4f5e-a418-5bcba288a3de)
